database-plugins/src/main/java/io/cdap/plugin/db/batch/source/DBSource.java

```
@@ -300,55 +303,56 @@ private String getBoundingQuery() {
-    private void validate() {
+    @SuppressWarnings("checkstyle:WhitespaceAround")
+    private void validate(FailureCollector collector) {
       boolean hasOneSplit = false;
-      if (!containsMacro("numSplits") && numSplits != null) {
+      if (!containsMacro(NUM_SPLITS) && numSplits != null) {
         if (numSplits < 1) {
-          throw new InvalidConfigPropertyException(
-            "Invalid value for numSplits. Must be at least 1, but got " + numSplits, "numSplits");
+          collector.addFailure("Invalid value for Number of Splits", "Must be at least 1").
+                  withConfigProperty(NUM_SPLITS);
         }
         if (numSplits == 1) {
           hasOneSplit = true;
         }
       }
 
-      if (!containsMacro("transactionIsolationLevel") && transactionIsolationLevel != null) {
-        TransactionIsolationLevel.validate(transactionIsolationLevel);
+      if (!containsMacro(TRANSACTION_ISOLATION_LEVEL) && transactionIsolationLevel != null) {
+        TransactionIsolationLevel.validate(transactionIsolationLevel, collector);
       }
 
       if (query != null) {
         return;
       }
 
-      if (!containsMacro("importQuery") && (query == null || query.isEmpty())
+      if (!containsMacro(IMPORT_QUERY) && (query == null || query.isEmpty())
         && (importQuery == null || importQuery.isEmpty())) {
-        throw new InvalidConfigPropertyException("An Import Query must be specified.", "importQuery");
+        collector.addFailure("Invalid import query specified", null).withConfigProperty(IMPORT_QUERY);
       }
 
-      if (!hasOneSplit && !containsMacro("importQuery") && !getImportQuery().contains("$CONDITIONS")) {
-        throw new InvalidConfigPropertyException(String.format("Import Query %s must contain the string '$CONDITIONS'.",
-                                                               importQuery), "importQuery");
+      if (!hasOneSplit && !containsMacro(IMPORT_QUERY) && !getImportQuery().contains("$CONDITIONS")) {
```

should be in else other wise can throw NPE if import query is not specified.